template/src/App.test.js

```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import { Provider } from "react-redux";
+import store from "./store";
+import App from "./App";
+
+test("renders learn react link", () => {
+  const { getByText } = render(
+    <Provider store={store}>
+      <App />
+    </Provider>
+  );
+  const linkElement = getByText(/learn redux/i);
+  expect(linkElement).toBeInTheDocument();
+});
```

Comment:
This could be a good opportunity to show an example of testing a simple Redux app, since it's more interactive than the default template. If we're just sticking with a basic smoke test this is fine with me.

Reply:
@markerikson any thoughts on this? I can do a more in-depth test if we want.

Reply:
Something like so? https://testing-library.com/docs/example-react-redux

Reply:
I have a suggestion:

```js

const setup = () =>
	render(
		<Provider store={store}>
			<App />
		</Provider>
	);

test('renders learn react link', () => {
	const { getByText } = setup();

	const linkElement = getByText(/redux toolkit/i);

	expect(linkElement).toBeInTheDocument();
});

test('can increment value', () => {
	const { getByTestId, getByText } = setup();
	fireEvent.click(getByText('+'));
	expect(getByTestId('count-value')).toHaveTextContent('1');
});

test('can decrement value', () => {
	const { getByTestId, getByText } = setup();
	fireEvent.click(getByText('-'));
	expect(getByTestId('count-value')).toHaveTextContent('0');
});
```

Reply:
I also believe it's worth to add tests for the reducer as well:

```js
// src/features/counter.test.js
import counter, { actions } from './counter';

test('can increment value', () => {
	expect(
		counter(
			{ value: 1 },
			{
				type: actions.increment.type,
			}
		).value
	).toEqual(2);
});

test('can decrement value', () => {
	expect(
		counter(
			{ value: 2 },
			{
				type: actions.decrement.type,
			}
		).value
	).toEqual(1);
});
```

Reply:
I can add something like that if we decide we want to add more in-depth testing.